addon/lib/toolbar-button.js

```diff
@@ -0,0 +1,159 @@
+      '32': iconPrefix + '-32.png',
+      '64': iconPrefix + '-64.png'
+    },
+    onClick: handleToolbarButtonClick
+  });
+
+  ToolbarButton.updateButtonBadge(); // eslint-disable-line no-use-before-define
+}
+
+function getExperimentList(availableExperiments, installedAddons) {
+  return Mustache.render(templates.experimentList, {
+    base_url: settings.BASE_URL,
+    view_all_params: getParams('view-all-experiments'),
+    experiments: Object.keys(availableExperiments).map(k => {
+      if (installedAddons[k]) {
+        availableExperiments[k].active = installedAddons[k].active;
+      }
+      availableExperiments[k].params = getParams(availableExperiments[k].title);
+      return availableExperiments[k];
+    })
+  });
+}
+
+function showExperimentList() {
+  panel.port.emit('show', getExperimentList(store.availableExperiments || {},
+                                            store.installedAddons || {}));
+}
+
+function getParams(title) {
+  return querystring.stringify({
+    utm_source: 'testpilot-addon',
+    utm_medium: 'firefox-browser',
+    utm_campaign: 'testpilot-doorhanger',
+    utm_content: title
+  });
+}
+
+function handleToolbarButtonClick() {
+  collapsed = !collapsed;
+  if (panel) panel.hide();
+  if (collapsed) return;
+
+  store.toolbarButtonLastClicked = Date.now();
+  ToolbarButton.updateButtonBadge(); // eslint-disable-line no-use-before-define
+
+  const experimentCount = ('availableExperiments' in store) ?
+    Object.keys(store.availableExperiments).length : 0;
+  panel.show({
+    width: PANEL_WIDTH,
+    height: (experimentCount * EXPERIMENT_HEIGHT) + FOOTER_HEIGHT,
+    position: button
+  });
+
+  // TODO: Record metrics event here, along with badge context
+}
+
+const ToolbarButton = module.exports = {
+
+  init: function(settingsIn) {
+    settings = settingsIn;
+
+    // update the our icon for devtools themes
+    Prefs.observe('devtools.theme', pref => {
+      setActionButton(pref === 'dark');
+    });
+    setActionButton(Prefs.get('devtools.theme') === 'dark');
+
+    collapsed = true; // collapsed state for panel
+    panel = Panel({ // eslint-disable-line new-cap
+      contentURL: './base.html',
+      contentScriptFile: './panel.js',
+      onHide: () => {
+        button.state('window', {checked: false});
+      }
+    });
+
+    panel.on('hide', () => collapsed = true);
+    panel.on('show', showExperimentList);
+    panel.port.on('back', showExperimentList);
+
+    panel.port.on('link', url => {
+      // TODO: Record metrics event here, along with badge context
+      tabs.open(url);
+      panel.hide();
+    });
+  },
+
+  destroy: function() {
+    panel.destroy();
+    button.destroy();
+  },
+
+  updateButtonBadge: function() {
+    // Bail if we haven't initialized the button yet.
+    if (!button) { return; }
+
+    // Initialize the last button click timestamp if necessary.
+    if (!('toolbarButtonLastClicked' in store)) {
+      store.toolbarButtonLastClicked = 1;
```

Comment:
@lmorchard I don't understand how this is setting a timestamp, shouldn't this be `new Date().getTime()`?